core/domain/feedback_jobs_continuous_test.py

```diff
@@ -696,3 +697,83 @@ def test_realtime_with_batch_computation(self):
                     'num_open_threads': 2,
                     'num_total_threads': 2,
                 })
+
+    def test_open_fixed_thread(self):
+        with self._get_swap_context():
+            # Create test objects.
+            user_id = 'uid'
+            exp_id = 'eid'
+            self.save_new_valid_exploration(exp_id, 'owner')
+
+            event_services.FeedbackThreadStatusChangedEventHandler.record(
+                exp_id, feedback_models.STATUS_CHOICES_FIXED,
+                feedback_models.STATUS_CHOICES_OPEN)
+
+            # Start job.
+            self.process_and_flush_pending_tasks()
+            MockFeedbackAnalyticsAggregator.start_computation()
+            self.assertEqual(
+                self.count_jobs_in_taskqueue(
+                    taskqueue_services.QUEUE_NAME_CONTINUOUS_JOBS), 1)
+            self.process_and_flush_pending_tasks()
+
+            # Stop job.
+            MockFeedbackAnalyticsAggregator.stop_computation(user_id)
+            self.assertEqual(
+                self.count_jobs_in_taskqueue(
+                    taskqueue_services.QUEUE_NAME_CONTINUOUS_JOBS), 0)
+
+            self._flush_tasks_and_check_analytics(
+                exp_id, {
+                    'num_open_threads': 0,
```

Comment:
What's going on here, conceptually?

Seems like _increment_open_threads_count() should be called. But in that case shouldn't num_open_threads be 1?

Reply:
First we are not creating any thread, that's why num_open_threads is zero. After this we create a new feedback thread and that's why num_open_threads gets incremented

Reply:
Doesn't this fall in the "status changed" category? https://github.com/oppia/oppia/blob/32aa0eee411d48798ce73ea1858fb2d0a1909d46/core/domain/feedback_jobs_continuous.py#L141

It seems like it should call increment_open_threads_count, and if there's no model, you'd expect num_open_threads to change to 1, while num_total_threads won't change. (Sometimes events may be received out of order.)

That doesn't seem to be happening here. Why is that?

Reply:
Changed the test. Is it OK now?

Reply:
I think it looks fine now, but I have one question. Can you explain why _flush_tasks_and_check_analytics() didn't give the correct result?

Reply:
_flush_tasks_and_check_analytics receive the values of num_open_threads and num_total_threads from FeedbackAnalyticsModel, not the real time model. And those values in FeedbackAnalyticsModel gets updated when we actually create a thread

Reply:
But if you look at get_thread_analytics_multi(), doesn't that combine the data from both the FeedbackAnalyticsModel and real-time model?